### PR TITLE
Latest update of file manager functionality in dapui

### PR DESCRIPTION
**Pull request by chuanlin2018**
Merged latest changes into integration. I added the new fontawesome library back into this branch because it was removed from oar-lib-angular so you will see lots of changes related to fontawesome. I can replace these library with latest fontawesome if necessary. 

I found following issues when testing but I think it related to DBIO backend:

1. Unable to save topics:
 
Console message:
customization.service.ts:170
       PUT https://localhost/midas/dap/mds3/mds3:0002/data/theme 405 (Data part not updatable)
 body ["Bioscience","Buildings and Construction"]
 
metadataupdate.service.ts:215 Failed to save metadata changes: server/system error:Unexpected Customization Error(405)
 
2. Save access page:
 
Console message:
PUT https://localhost/midas/dap/mds3/mds3:0002/data/pdr:see 400 (Invalid Input Data)
 @Body:
{"components":[{"@type":["nrdp:AccessPage"],"isNew":true,"accessURL":https://dx.doi.org/10.6028/NIST.TN.1913,"title":"test","description":"test"},{"_schema":https://data.nist.gov/od/dm/nerdm-schema/v0.7#/definitions/Component,"@context":https://data.nist.gov/od/dm/nerdm-pub-context.jsonld,"@id":"83","@type":["nrdp:DataFile","nrdp:DownloadableFile","dcat:Distribution"],"_extensionSchemas":[https://data.nist.gov/od/dm/nerdm-schema/pub/v0.7#/definitions/DataFile],"filepath":"image001.png","downloadURL":"pdr:nrd:@id/nrd:filepath","size":15850,"checksum":{"hash":"a4d1aac1402462f3c214b8b315aa257f2d1b40075dfec3bf0c5eb3c8b3371438","algorithm":{"@type":"Thing","tag":"sha256"}}}]}

3. When I delete the last file in NextCloud and click on refresh files in landing page, the file was not removed from the list.

### CODE REVIEW

**Comment by chuanlin2018** [comment]
Changes in the latest checked in:

1. When user clicked on "Finish" button in Wizard, the landing page will show up in the same tab.
2. The "Manage Files" button at the top of the landing page is now working.
3. Access page can now be saved (previous version put the whole component object instead of just access page).
4. Fixed unit test.

File: oar-lps/libs/oarlps/src/lib/landing/data-files/data-files.component.ts [hunk 1]
```diff
@@ -830,31 +848,64 @@ export class DataFilesComponent implements OnInit, OnChanges {
...
-        })
+            this.mdupdsvc.loadDraft(true).subscribe({
+                next: (md) => 
+                {
+                    if(md)
+                    {
+                        this.mdupdsvc.setOriginalMetadata(md as NerdmRes);
+                        this.mdupdsvc.checkUpdatedFields(md as NerdmRes);
+
+                        this.record['components'] = JSON.parse(JSON.stringify(md['components']));
```
Comment (RayPlante) [inline oar-lps/libs/oarlps/src/lib/landing/data-files/data-files.component.ts:869]: This line throws an exception when `md['components']` is undefined (i.e. when `md` does not have this property).  Note that the `components` property is optional in NERDm.  In particular, this happens when the record has just been created.  Clicking the "Refresh Files" button is pressed when just after creation, before any files have been put into the file manager space.  

Consider something like this:
```javascript
                        if (md['components']) 
                            this.record['components'] = JSON.parse(JSON.stringify(md['components']));
                        else
                            this.record['components'] = []
```

**Review by RayPlante — CHANGES_REQUESTED** [review]

File: oar-lps/libs/oarlps/src/lib/landing/data-files/data-files.component.ts (hunk 1 above)
Comment (chuanlin2018) [reply oar-lps/libs/oarlps/src/lib/landing/data-files/data-files.component.ts:869]: Sure. Will update soon...

**Comment by chuanlin2018** [comment]
Updated.

**Comment by RayPlante** [comment]
that works!

**Comment by chuanlin2018** [comment]
Awesome!

**Review by RayPlante — APPROVED** [review]
Fixes work--deployed to midasdev.

**Pull request merged**